code/graphics/software/font.cpp

```diff
@@ -95,6 +95,27 @@ namespace
 			}
 		}
 
+		if (optional_string("+Auto Size:")) {
+			bool autoSize;
+			stuff_boolean(&autoSize);
+
+			if (autoSize) {
+				// Lambda to calculate font size based on screen resolution
+				auto calculateAutoSize = [size]() -> float {
+					int vmin = std::min(gr_screen.max_w, gr_screen.max_h);
+
+					// Base size calculation (similar to ~Npx font at 1080p)
+					// Use 1080p because that's generally what font sizes have been targeting for years
+					// And should provide a fairly close out-of-the-box solution
+					float baseSize = vmin * (size / 1080.0f);
```

Comment:
Just thinking about this a bit more, I reckon you might have also thought about making this a game settings value that defaults to 1080?

Reply:
Game settings isn't the right place because this is a player thing, not a game thing. This PR really should be merged after #6330 so that players can adjust the font from there but in my googling and tests this formula gets the font size roughly close enough.

Reply:
Ah right, that makes sense, thanks!